dubbo-common/src/main/java/org/apache/dubbo/common/utils/PojoUtils.java

```diff
@@ -1,628 +1,1012 @@
+                    forceSetFieldValue(field, dest, value);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Failed to set field " + name + " of pojo " + dest.getClass().getName() + " : " + e.getMessage(), e);
+                }
+                return true;
+            }
+        }
+        if(method == null) {
+            return false;
+        }
+        GenericFeature genericFeature = method.getAnnotation(GenericFeature.class);
+        if(genericFeature == null) {
+            return false;
+        }
+        if (genericFeature.ignore()) {
+            return true;
+        }
+
+        if (!method.isAccessible()) {
+            method.setAccessible(true);
+        }
+        if(genericFeature.nullNotRealize() && value == null) {
+            return true;
+        }
+        if (value instanceof String) {
+            value = fromValueType(value, method.getParameterTypes()[0], genericFeature);
+        }
+        if(rawValue == value) {
+            Type ptype = method.getGenericParameterTypes()[0];
+            value = realize0(value, method.getParameterTypes()[0], ptype, history);
+        }
+
+        try {
+            method.invoke(dest, value);
+        } catch (Exception e) {
+            String exceptionDescription = "Failed to set pojo " + dest.getClass().getSimpleName() + " property " + name
+                    + " value " + value + "(" + value.getClass() + "), cause: " + e.getMessage();
+            logger.error(exceptionDescription, e);
+            throw new RuntimeException(exceptionDescription, e);
+        }
+        return true;
+
+    }
+
+    private static void forceSetFieldValue(Field field,Object dest,Object value) throws IllegalAccessException {
+        boolean isNotPublic = Modifier.isPublic(field.getModifiers());
+        if(!isNotPublic) {
+            field.setAccessible(true);
+        }
+        try{
+            field.set(dest,value);
+        } finally {
+            if(!isNotPublic) {
+                field.setAccessible(false);
+            }
+        }
+    }
+
+    /**
+     * Get parameterized type
+     *
+     * @param genericType generic type
+     * @param index       index of the target parameterized type
+     * @return Return Person.class for List<Person>, return Person.class for Map<String, Person> when index=0
+     */
+    private static Type getGenericClassByIndex(Type genericType, int index) {
+        Type clazz = null;
+        // find parameterized type
+        if (genericType instanceof ParameterizedType) {
+            ParameterizedType t = (ParameterizedType) genericType;
+            Type[] types = t.getActualTypeArguments();
+            clazz = types[index];
+        }
+        return clazz;
+    }
+
+    private static Object newInstance(Class<?> cls) {
+        try {
+            return cls.newInstance();
+        } catch (Throwable t) {
+            try {
+                Constructor<?>[] constructors = cls.getDeclar
```

Comment:
Is it better to have a cache for this method?And i think it is better in `ReflectUtils.class`